### PR TITLE
Add VanillaGAN tensorflow files

### PR DESCRIPTION
**Pull request by ashishmurali**
Add VanillaGAN tensorflow files

### CODE REVIEW

File: generative_models/VanillaGAN_TensorFlow/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# TensorFlow Implementation of VanillaGAN on MNIST Dataset
+
+### Usage
+```bash
+$ python3 main.py --epochs 50 --batch_size 128 --outdir 'desired location in system'
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/README.md]:
```suggestion
$ python3 main.py --epochs 50 --batch_size 128 --outdir "." 
```

File: generative_models/VanillaGAN_TensorFlow/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# TensorFlow Implementation of VanillaGAN on MNIST Dataset
+
+### Usage
+```bash
+$ python3 main.py --epochs 50 --batch_size 128 --outdir 'desired location in system' 
+```
+NOTE: on Colab Notebook use following command:
+```bash
+!git clone link-to-repo
+%run main.py --epochs 50 --batch_size 128 --outdir 'desired location in system'
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/README.md]:
```suggestion
%run main.py --epochs 50 --batch_size 128 --outdir "."
```

File: generative_models/VanillaGAN_TensorFlow/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+```
+
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/README.md:30]: add help log, copy the output of `python3 main.py --help`

File: generative_models/VanillaGAN_TensorFlow/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+* **Title**: Generative Adversarial Networks
+* **Authors**: Ian J. Goodfellow, Jean Pouget-Abadie, Mehdi Mirza, Bing Xu, David Warde-Farley, Sherjil Ozair, Aaron Courville, Yoshua Bengio
+* **Link**: http://arxiv.org/abs/1406.2661
+* **Tags**: Neural Network, GAN, generative models, unsupervised learning
+* **Year**: 2014
+
+# Summary
+
+* What
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/README.md]: may be add complete sentence

File: generative_models/VanillaGAN_TensorFlow/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  * GANs are based on adversarial training.
+  * Adversarial training is a basic technique to train generative models (so here primarily models that create new images).
+  * In an adversarial training one model (G, Generator) generates things (e.g. images). Another model (D, discriminator) sees real things (e.g. real images) as well as fake things (e.g. images from G) and has to learn how to differentiate the two.
+  * Neural Networks are models that can be trained in an adversarial way (and are the only models discussed here).
+
+* How
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/README.md]: complete sentence

File: generative_models/VanillaGAN_TensorFlow/models.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import tensorflow as tf
+from tensorflow.keras.layers import Dense, Dropout, Input, BatchNormalization,LeakyReLU
+from tensorflow.keras.models import Model,Sequential
+from tensorflow.keras.optimizers import Adam
+
+def adam_optimizer():
+    return Adam(lr=0.0002, beta_1=0.5)
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/models.py]: add learning rate as parser arg

Comment (nirmal-suthar) [reply generative_models/VanillaGAN_TensorFlow/models.py]: and beta_1 too, with default as 0.5

File: generative_models/VanillaGAN_TensorFlow/models.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+def create_generator():
+
+    generator=Sequential()
+
+    generator.add(Dense(units=256,input_dim=100))
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/models.py]: add encoding_dims as parser args and change according to arg instead of hardcoding it to 100

File: generative_models/VanillaGAN_TensorFlow/main.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+          discriminator.trainable=False
+          g_loss = gan.train_on_batch(noise,valid)
+
+          print ("%d [D loss: %f] [G loss: %f]" % (e, d_loss, g_loss))
+        if ipython:
+            display.clear_output(wait=True)
+        plot_generated_images(e, generator,seed,outdir)
+
+
+
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/main.py]: remove stray lines

File: generative_models/VanillaGAN_TensorFlow/main.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    batch_count = int(X_train.shape[0] / batch_size)
+
+    generator= create_generator()
+    discriminator= create_discriminator()
+    gan = create_gan(discriminator, generator)
+
+    valid = np.ones((batch_size, 1))
+    fake = np.zeros((batch_size, 1))
+
+    seed = np.random.normal(0,1, [25, 100])
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/main.py]: take 100 from encoding_dims arg

File: generative_models/VanillaGAN_TensorFlow/main.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+    for e in range(1,epochs+1 ):
+        print("Epoch %d" %e)
+        for _ in range(batch_size):
+
+          noise= np.random.normal(0,1, [batch_size, 100])
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/main.py]: 100 from encoding_dims arg

File: generative_models/VanillaGAN_TensorFlow/main.py [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+          generated_images = generator.predict(noise)
+
+          image_batch = X_train[np.random.randint(low=0,high=X_train.shape[0],size=batch_size)]
+
+          discriminator.trainable=True
+          d_loss_real = discriminator.train_on_batch(image_batch, valid)
+          d_loss_fake = discriminator.train_on_batch(generated_images, fake)
+          d_loss = 0.5 * np.add(d_loss_real, d_loss_fake)
+
+          noise= np.random.normal(0,1, [batch_size, 100])
```
Comment (nirmal-suthar) [inline generative_models/VanillaGAN_TensorFlow/main.py]: change 100

**Review by nirmal-suthar — CHANGES_REQUESTED** [review]

**Comment by nirmal-suthar** [comment]
Also, it would be great if you could compress gif file a bit (approx to 1.5mb)

File: generative_models/VanillaGAN_TensorFlow/README.md (hunk 1 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/README.md]: done

File: generative_models/VanillaGAN_TensorFlow/README.md (hunk 2 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/README.md]: done

File: generative_models/VanillaGAN_TensorFlow/README.md (hunk 3 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/README.md:30]: done

File: generative_models/VanillaGAN_TensorFlow/README.md (hunk 4 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/README.md]: applied some changes

File: generative_models/VanillaGAN_TensorFlow/README.md (hunk 5 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/README.md]: applied some changes

File: generative_models/VanillaGAN_TensorFlow/models.py (hunk 6 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/models.py]: done

File: generative_models/VanillaGAN_TensorFlow/models.py (hunk 7 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/models.py]: done

File: generative_models/VanillaGAN_TensorFlow/main.py (hunk 8 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/main.py]: done

File: generative_models/VanillaGAN_TensorFlow/main.py (hunk 9 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/main.py]: done

File: generative_models/VanillaGAN_TensorFlow/main.py (hunk 10 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/main.py]: done

File: generative_models/VanillaGAN_TensorFlow/main.py (hunk 11 above)
Comment (ashishmurali) [reply generative_models/VanillaGAN_TensorFlow/main.py]: done
